holoviews/plotting/widgets/__init__.py

```diff
@@ -106,9 +107,8 @@ def __init__(self, plot, renderer=None, **params):
         super(NdWidget, self).__init__(**params)
         self.id = plot.comm.target if plot.comm else uuid.uuid4().hex
         self.plot = plot
-        self.dimensions = plot.dimensions
-        self.keys = plot.keys
-
+        dims, keys = drop_streams(plot.streams, plot.keys, plot.dimensions)
+        self.dimensions, self.keys = dims, keys
```

Comment:
As the variables ``dims`` and ``keys`` don't seem to be used in this method for anything other than setting ``self.dimensions`` and ``self.keys``, I would consider using:

```python
self.dimensions, self.keys = drop_streams(plot.streams, 
                                          plot.keys, 
                                          plot.dimensions)